### PR TITLE
[OVERALL FIXING] Fixed some issues related to Payment and Rental

### PR DESCRIPTION
**Pull request by ivan0dyatlyukkk**

### CODE REVIEW

**Review by OleksandrSaven — APPROVED** [review]

File: src/main/java/com/project/carsharingapp/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -76,6 +77,13 @@ public User getByEmail(String email) {
         );
     }
 
+    @Override
+    public User getByAuthentication(Authentication auth) {
```
Comment (DmytroV95) [inline src/main/java/com/project/carsharingapp/service/impl/UserServiceImpl.java:83]:
```suggestion
    public User getAuthenticatedUser(Authentication auth) {
```

File: src/main/java/com/project/carsharingapp/service/payment/PaymentServiceImpl.java [hunk 2]
```diff
@@ -35,14 +35,15 @@ public class PaymentServiceImpl implements PaymentService {
...
+        Rental rental = getUserRentalById(authentication, requestDto.getRentalId());
         Payment.Type type = Payment.Type.valueOf(requestDto.getType());
 
+        checkIfPaymentSessionValid(rental);
+
+        if (paymentRepository.existsByRentalIdAndStatus(rental.getId(), Payment.Status.PAUSED)) {
+            return paymentMapper.toDto(paymentRepository.findByRentalId(rental.getId()).get());
+        }
+
         try {
```
Comment (DmytroV95) [inline src/main/java/com/project/carsharingapp/service/payment/PaymentServiceImpl.java:51]: Redundant lines

File: src/main/java/com/project/carsharingapp/service/payment/PaymentServiceImpl.java [hunk 3]
```diff
@@ -63,6 +64,8 @@ public PaymentResponseDto updateStatus(String sessionId, Payment.Status status)
 
     @Override
     public List<PaymentResponseDto> getAll(Authentication authentication, Pageable pageable) {
+        User user = userService.getByAuthentication(authentication);
+
```
Comment (DmytroV95) [inline src/main/java/com/project/carsharingapp/service/payment/PaymentServiceImpl.java:74]: Redundant line

File: src/main/java/com/project/carsharingapp/service/payment/PaymentServiceImpl.java [hunk 4]
```diff
@@ -75,7 +78,22 @@ private Payment generatePayment(Session session, Rental rental, Payment.Type typ
...
+    private Rental getUserRentalById(Authentication authentication, Long id) {
+        User user = userService.getByAuthentication(authentication);
+        return rentalService.getByUserAndId(user, id);
+    }
+
+    private void checkIfPaymentSessionValid(Rental rental) {
+        if (paymentRepository.existsByRentalIdAndStatus(rental.getId(), Payment.Status.PAID)) {
+            throw new NotValidPaymentProcessException("The rental is paid!");
+        }
+
```
Comment (DmytroV95) [inline src/main/java/com/project/carsharingapp/service/payment/PaymentServiceImpl.java]: Redundant line

File: src/main/java/com/project/carsharingapp/service/impl/RentalServiceImpl.java [hunk 5]
```diff
@@ -53,6 +53,14 @@ public List<RentalDto> getByUserIdAndActiveStatus(Pageable pageable,
                 .toList();
     }
 
+    @Override
+    public Rental getByUserAndId(User user, Long id) {
```
Comment (DmytroV95) [inline src/main/java/com/project/carsharingapp/service/impl/RentalServiceImpl.java:78]: Can you specify in method name by what entity id you want to get?

**Review by DmytroV95 — CHANGES_REQUESTED** [review]
GJ! Left some comments.

**Review by AntonZhdanov — APPROVED** [review]

**Review by OlegStashkiv — APPROVED** [review]

**Pull request merged**